src/agent/src/network.rs

```diff
@@ -28,3 +37,121 @@ impl Network {
         self.dns.push(dns);
     }
 }
+
+pub fn setup_guest_dns(logger: Logger, dns_list: Vec<String>) -> Result<Vec<String>> {
```

Comment:
Why we pass the dns_list as the parameter and then return it as the return value?  If the parameter and the return value is the same, why not only return Result<()> instead?

Reply:
Yes, it's not needed.